### DISCUSSION TITLE
Navigation module ignoring @navigation.register decorator

### DISCUSSION THREAD

**Discussion by RicardoFernandez-UY** [anvilistas/anvil-extras#451] [Q&A] [answered]
Hi,

I’m trying to use the anvil-extras Navigation together with the Hash Routing navigation method.
I use the Navigation module for generating the menus dinamically, and I want to display the Form titles using its functionallity, in order to display a different title for each form in the Title slot of the template form (the one that holds the menu).

Here is what I’ve done:

I started with the Hash Routing only. Everything worked ok.
Then, I introduced the Navigation module, and made the required changes.

At this point the Navigation module is taking the task by generating the menus and the events needed to navigate to the target forms (set_url_hash(sender.tag.target)), as stucork explained to my in another post. 
But the Navigation module still needs the @routing.route from the Hash Navigation, to be present. Otherwise it does not work.

Here comes my confusion about how this is working at all, because per documentation:

```
In order for a form to act as a target of a menu link, it has to register a name with the navigation module using a decorator on its class definition.
For example:
@navigation.register(name=“about”, title=“About”)
```

And I’m not using @navigation.register anywhere.
I did try using it, because I wanted to display the Title of the Form, but it completely ignores it.

So, I’m clearly missing something here. What could that be?
In summary, how is this navigating to the target forms if I haven’t registered them with the Navigation module?, and why when I use the @navigation.register, it ignores it and so the title functionallity?

```
Here is the link to a minimum version of the app.
https://anvil.works/build#clone:E7LKCQ22EHWX64DL=LL6ZM2STKEW2HBOYJ22K5C6E
```
You will see the menu items defined in the visual designer, which are the ones I used for working directly with the Hash Routing + nav_click event, at the beginning, and also the dinamically generated links  (the ones with icos) using the Navigation menu, which are the ones that are working now.

Thank you !!!

**Comment by s-cork** [comment, marked as answer]
Looking at the source code. The navigation.register has no effect when using hash routing. Which also means the title part of register is totally ignored.

So might worth writing an issue for this as a bug or feature request.

**Comment by meatballs** [comment]
Yep. Hash routing was added to the navigation module as a bit of an afterthought.

Navigation includes a bespoke function for handling the title etc. when using "classic" routing, but it just delegates entirely to hash routing when that's being used and the decorator has no effect at all.

**Reply by RicardoFernandez-UY** [reply]
Hi!
Okey. So, mystery solved ;-)
I'm glad I contributed to at least finding it. On the other hand, I'm sorry I cannot add to much to the solution.
Can you think of a workaround to set the Title while using the hash routing with the navigation? (because these two modules are a great combo, and being able to display the title would be the icing on the cake)
Thank you very much!!

**Reply by s-cork** [reply]
You could do this in the on_navigation callback to do that.
Take a look at the routing docs to see how this might be used. 

You could also look at the source code for the navigation module and create your own build_menu function.

**Reply by RicardoFernandez-UY** [reply]
Ok, thank you; I'll give it a try. 
Unfortunately I'm not much of a programmer, but I'll do my best, and in case I end with something usable I'll share it here.
